experiment/nursery/create_bazel_cache_rcs.sh

```diff
@@ -29,6 +33,11 @@ make_bazel_rc () {
     echo "build --remote_http_cache=${CACHE_HOST}" >> $1
     # don't fail if the cache is unavailable
     echo "build --remote_local_fallback" >> $1
+    # make sure the cache considers host toolchain versions
+    GCC_VERSION=$(package_version gcc)
+    echo "build --action_env=__CACHE_GCC_VERSION__=${GCC_VERSION}" >> $1
+    PYTHON_VERSION=$(package_version python)
+    echo "build --action_env=__CACHE_PYTHON_VERSION__=${PYTHON_VERSION}" >> $1
```

Comment:
FYI if you want to stop writing `echo ... >> $1` all the time, you can probably do something like 
```bash

cat >$1 <<EOF
# this is the default for recent releases but we set it explicitly
# since this is the only hash our cache supports
startup --host_jvm_args=-Dbazel.DigestFunction=sha256
# use remote caching for all the things
build --experimental_remote_spawn_cache
# point it at our http cache ...
build --remote_http_cache=${CACHE_HOST}
...
EOF
```

Reply:
hmm the downside to this is that I have to deindent the block being written and the whole block highlights as a string :man_shrugging:

Reply:
other alternatives:

1. wrap everything in a subshell `( )`, e.g.
```shell
(
  echo "blah"
) > $1
```
2. Rather than having `make_bazel_rc` take an argument, just pipe it to a file below, e.g.
```shell
make_bazel_rc > "${HOME}/.bazelrc"
make_bazel_rc > /etc/bazel.bazelrc

Reply:
(don't consider these comments blocking, though)

Reply:
Thanks! I'll go with one of these. Blocking this myself on verifying that I can disrupt the cache with GCC version locally and then fix it with these args. Will do one of these first.

Reply:
done